src/webgpu/web_platform/canvas/readbackFromWebGPUCanvas.spec.ts

```diff
@@ -81,12 +84,32 @@ async function initCanvasContent(
     size: rows * bytesPerRow,
     usage: GPUBufferUsage.COPY_SRC,
   });
+  const blue: { [index: string]: Uint8Array } = {
+    bgra8unorm: new Uint8Array([0xff, 0x00, 0x00, 0xff]),
+    rgba8unorm: new Uint8Array([0x00, 0x00, 0xff, 0xff]),
+  };
+
+  const green: { [index: string]: Uint8Array } = {
+    bgra8unorm: new Uint8Array([0x00, 0xff, 0x00, 0xff]),
+  };
+  green['rgba8unorm'] = green['bgra8unorm'];
+
+  const red: { [index: string]: Uint8Array } = {
+    bgra8unorm: new Uint8Array([0x00, 0x00, 0xff, 0xff]),
+    rgba8unorm: new Uint8Array([0xff, 0x00, 0x00, 0xff]),
+  };
+
+  const yellow: { [index: string]: Uint8Array } = {
+    bgra8unorm: new Uint8Array([0x00, 0xff, 0xff, 0xff]),
+    rgba8unorm: new Uint8Array([0xff, 0xff, 0x00, 0xff]),
+  };
+
   const mapping = buffer.getMappedRange();
   const data = new Uint8Array(mapping);
-  data.set(new Uint8Array([0xff, 0x00, 0x00, 0xff]), 0); // blue
-  data.set(new Uint8Array([0x00, 0xff, 0x00, 0xff]), 4); // green
-  data.set(new Uint8Array([0x00, 0x00, 0xff, 0xff]), 256 + 0); // red
-  data.set(new Uint8Array([0x00, 0xff, 0xff, 0xff]), 256 + 4); // yellow
+  data.set(blue[format], 0); // blue
+  data.set(green[format], 4); // green
+  data.set(red[format], 256 + 0); // red
+  data.set(yellow[format], 256 + 4); // yellow
   buffer.unmap();
```

Comment:
Perhaps change this to clear the texture using a render pass clear color, which doesn't care what the format is. That extends more easily (don't need the blue/green/red/yellow constant dictionaries anymore) and to (rgba16float, rgb10a2unorm).

Reply:
I'd still prefer to do this, but OK if you think we shouldn't.